src/components/ChallengeModal/ChallengeModal.tsx

```diff
@@ -55,6 +62,33 @@ interface ChallengeModalProperties {
     playersList?: Array<{ name: string; rank: number }>;
     tournamentRecordId?: number;
     tournamentRecordRoundId?: number;
+    created?: (c: ChallengeDetails) => void;
+}
+
+interface ChallengeState {
+    live?: boolean; // computed during creation of challenge
+    initialized: boolean;
+    min_ranking: number;
+    max_ranking: number;
+    challenger_color: string; // "automatic"
+    game: {
+        name: string; //"",
+        rules: string; // "japanese",
+        ranked: boolean;
+        width: IdType;
+        height: IdType;
+        handicap: number; // 0,
+        komi_auto: string; //"automatic",
+        komi: number; // 5.5,
+        disable_analysis: boolean;
+        initial_state: any; // TBD
+        private: boolean;
+        rengo: boolean;
+        rengo_casual_mode: boolean;
+        pause_on_weekends: boolean;
+        time_control: TimeControlTypes.TimeControlSystem;
+        time_control_parameters: TimeControlPicker;
```

Comment:
Shouldn't `time_control_parameters` be a `TimeControl`?

Reply:
I'm basing that off this line:

https://github.com/online-go/online-go.com/blob/6a5c8a15f6a570d0c128dd9d153fb6299c550aa8/src/components/ChallengeModal/ChallengeModal.tsx#L493

`this.refs.time_control_picker` should be the `TimeControlPicker`, so `this.refs.time_control_picker.time_control` should be a `TimeControl`.

Reply:
Dad gang.  Another "so.... how does this even work then?!"

Reply:
JS helpfully guessing what the monkey meant to do, even if he didn't do it ❤️

Reply:
Kinda more like JS doing what it always does, regardless of what we're telling the compiler...

Reply:
So I've totally refactored everything again.  Standby :)